src/modules/git_branch.rs

```diff
@@ -31,29 +31,37 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 
     let repo = context.get_repo().ok()?;
     let branch_name = repo.branch.as_ref()?;
-    let truncated_graphemes = get_graphemes(&branch_name, len);
-    // The truncation symbol should only be added if we truncated
-    let truncated_and_symbol = if len < graphemes_len(&branch_name) {
-        truncated_graphemes + &truncation_symbol
+
+    let mut graphemes = get_graphemes(&branch_name);
+    let trunc_len = len.min(graphemes.len());
+
+    if trunc_len < graphemes.len() {
+        // The truncation symbol should only be added if we truncate
+        graphemes[trunc_len] = truncation_symbol;
+        graphemes.truncate(trunc_len + 1)
+    }
+
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter.map(|variable| match variable {
+            "branch" => Some(graphemes.concat()),
+            _ => None,
+        })
     } else {
-        truncated_graphemes
+        log::error!("Error parsing format string in `git_branch.format`");
+        return None;
```

Comment:
API has changed in upstream `feat/format-string` branch.

Here is an example: https://github.com/starship/starship/pull/1063/files

Reply:
Thanks, I have rebased against that branch.